### ISSUE TITLE
shopify/jsx-no-hardcoded-content not linting correctly

### ISSUE THREAD

**Issue by utkarshsaxena93** [Shopify/eslint-plugin-shopify#81]
```
"shopify/jsx-no-hardcoded-content": ['error', {
        "allowStrings": false,
        "allowNumbers": false
      }]
```
does not lint correctly. It throws and error for all hardcoded values in JSX. 

Example:
I see the following error:
`9:5  error  Do not use hardcoded content as the children of the div component  shopify/jsx-no-hardcoded-content`

And this is my code 
```
function Preview() {
  return (
    <div className={styles.Preview}>
      <div className={styles.Content} />
    </div>
  );
}
```

Similarly there are many such cases where it shouldn't throw an error

**Comment by utkarshsaxena93** [comment]
c/c @lemonmade 
best way to repro this is to enable it in web repo and try to look at files for which it is complaining. You should see instances where it shouldn't be complaining.
